### DISCUSSION TITLE
Make Menu from example to close on hover out

### DISCUSSION THREAD

**Discussion by adenisovgit** [floating-ui/floating-ui#2064] [Q&A]
Hi.
Original Menu from example could be opened and closed only by clicks.
I changed a little bit the useHover condition 
https://codesandbox.io/s/misty-sun-3gkfgw?file=/src/DropdownMenu.tsx
line 86
from 
        `enabled: nested && allowHover,`
to
       ` enabled: (nested || (!nested && open)) && allowHover,`

and now it is opens on click and closing on hover out.
But there is conflict on closing.
If you just click root menu to open|close keeping still it will close normally. 
But if you click to open then move a little and then click again - this next click will not close the menu, just the next one will close.

I solved this by changing useClick from mousedown to click (line 92) and changing onClick
line 168
from
```
          onClick(event) {
            event.stopPropagation();
          },
```
to
```
            onClick(event) {
              event.stopPropagation();
              // next to solve conflict between hover and click closing rootMenu
              if (!nested && open) {
                setOpen(false);
              }
            },
```
but I am sure there is more elegant solution combining useHover and useClick.
Could somebody help with that?

**Comment by atomiks** [comment]
The CodeSandbox doesn't contain the code as described here

**Reply by adenisovgit** [reply]
now it does. I thought it has autosave

**Reply by adenisovgit** [reply]
now there is version with lost click to close on codesandbox

**Reply by adenisovgit** [reply]
@atomiks any suggestions? Actually, it looks like a bug.
